src/rpc/RPCHelpers.cpp

```diff
@@ -1478,4 +1478,210 @@ specifiesCurrentOrClosedLedger(boost::json::object const& request)
+    {
+        if (request.contains(JS(ledger_index_max)) ||
+            request.contains(JS(ledger_index_min)))
+            return Status{
+                Error::rpcINVALID_PARAMS, "containsLedgerSpecifierAndRange"};
+
+        auto v = ledgerInfoFromRequest(context);
+        if (auto status = std::get_if<Status>(&v))
+            return *status;
+
+        maxIndex = minIndex = std::get<ripple::LedgerInfo>(v).seq;
+    }
+
+    if (!cursor)
+    {
+        if (forward)
+            cursor = {minIndex, 0};
+        else
+            cursor = {maxIndex, INT32_MAX};
+    }
+
+    std::uint32_t limit;
+    if (auto const status = getLimit(context, limit); status)
+        return status;
+
+    if (request.contains(JS(limit)))
+        response[JS(limit)] = limit;
+
+    boost::json::array txns;
+    auto [blobs, retCursor] = getter(limit, forward, cursor);
+    auto dbFetchEnd = std::chrono::system_clock::now();
+
+    if (retCursor)
+    {
+        boost::json::object cursorJson;
+        cursorJson[JS(ledger)] = retCursor->ledgerSequence;
+        cursorJson[JS(seq)] = retCursor->transactionIndex;
+        response[JS(marker)] = cursorJson;
+    }
+
+    std::optional<size_t> maxReturnedIndex;
+    std::optional<size_t> minReturnedIndex;
+    for (auto const& txnPlusMeta : blobs)
+    {
+        if (txnPlusMeta.ledgerSequence < minIndex ||
+            txnPlusMeta.ledgerSequence > maxIndex)
+        {
+            BOOST_LOG_TRIVIAL(debug)
+                << __func__
+                << " skipping over transactions from incomplete ledger";
+            continue;
+        }
+
+        boost::json::object obj;
+
+        if (!binary)
+        {
+            auto [txn, meta] = toExpandedJson(txnPlusMeta);
+            obj[JS(meta)] = meta;
+            obj[JS(tx)] = txn;
+            obj[JS(tx)].as_object()[JS(ledger_index)] =
+                txnPlusMeta.ledgerSequence;
+            obj[JS(tx)].as_object()[JS(date)] = txnPlusMeta.date;
+        }
+        else
+        {
+            obj[JS(meta)] = ripple::strHex(txnPlusMeta.metadata);
+            obj[JS(tx_blob)] = ripple::strHex(txnPlusMeta.transaction);
+            obj[JS(ledger_index)] = txnPlusMeta.ledgerSequence;
+            obj[JS(date)] = txnPlusMeta.date;
+        }
+        obj[JS(validated)] = true;
+
+        txns.push_back(obj);
+        if (!minReturnedIndex || txnPlusMeta.ledgerSequence < *minReturnedIndex)
+            minReturnedIndex = txnPlusMeta.ledgerSequence;
+        if (!maxReturnedIndex || txnPlusMeta.ledgerSequence > *maxReturnedIndex)
+            maxReturnedIndex = txnPlusMeta.ledgerSequence;
+    }
+
+    response[JS(ledger_index_min)] = minIndex;
+    response[JS(ledger_index_max)] = maxIndex;
+
+    response[JS(transactions)] = txns;
+
+    auto serializationEnd = std::chrono::system_clock::now();
+    BOOST_LOG_TRIVIAL(info)
+        << __func__ << " serialization took "
+        << ((serializationEnd - dbFetchEnd).count() / 1000000000.0);
```

Comment:
Let's remove the hard coded number here, and change all of these to:
`std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count()`

I think this is the syntax, but double check me.